File: LHDS.AdminPortal.Api.Tests.Acceptance/Models/DataSets/Dataset.cs
// ---------------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------------

using System;

namespace LHDS.AdminPortal.Api.Tests.Acceptance.Models.DataSets
{
    public class DataSet
    {
        public Guid Id { get; set; }
        public string DataSetName { get; set; }
        public string DataSetAliasses { get; set; }
        public string DataSetSupplier { get; set; }
        public string DataSetAuthor { get; set; }
        public string SpecifiedBy { get; set; }
        public bool IsNationallySpecified { get; set; }
        public string CollectedBy { get; set; }
        public bool IsNationallyCollected { get; set; }
        public string DataSourceType { get; set; }
        public bool IsActive { get; set; }
        public DateTimeOffset? ActiveFrom { get; set; }
        public DateTimeOffset? ActiveTo { get; set; }
        public string CreatedBy { get; set; }
        public string UpdatedBy { get; set; }
        public DateTimeOffset UpdatedDate { get; set; }
        public DateTimeOffset CreatedDate { get; set; }

    }
}

File: LHDS.AdminPortal.Configurations/lhds.adminportal.configurations.client/src/models/dataSets/dataSet.ts
export class DataSet {
    public id: string;
    public dataSetName: string;
    public dataSetAliases: string;
    public dataSetSupplier: string;
    public dataSetAuthor: string;
    public specifiedBy: string;
    public IsNationallySpecified: boolean;
    public collectedBy: string;
    public isNationallyCollected: boolean;
    public dataSourceType: string;
    public isActive: boolean;
    public activeFrom?: Date;
    public activeTo?: Date;
    public createdBy?: string;
    public createdDate?: Date;
    public updatedBy?: string;
    public updatedDate?: Date;

    constructor(dataSet: DataSet) {
        this.id = dataSet.id ? dataSet.id : "";
        this.dataSetName = dataSet.dataSetName || "";
        this.dataSetAliases = dataSet.dataSetAliases || "";
        this.dataSetSupplier = dataSet.dataSetSupplier || "";
        this.dataSetAuthor = dataSet.dataSetAuthor || "";
        this.specifiedBy = dataSet.specifiedBy || "";
        this.IsNationallySpecified = dataSet.IsNationallySpecified === true ? true : false;
        this.collectedBy = dataSet.collectedBy || "";
        this.isNationallyCollected = dataSet.isNationallyCollected === true ? true : false;
        this.dataSourceType = dataSet.dataSourceType || "";
        this.isActive = dataSet.isActive === true ? true : false;
        this.activeFrom = dataSet.activeFrom;
        this.activeTo = dataSet.activeTo;
        this.createdDate = dataSet.createdDate;
        this.createdBy = dataSet.createdBy;
        this.createdDate = new Date(dataSet.createdDate);
        this.updatedBy = dataSet.updatedBy;
        this.updatedDate = new Date(dataSet.updatedDate);
    }
}